src/System.Private.CoreLib/shared/System/Diagnostics/Tracing/CounterGroup.cs

```diff
@@ -205,48 +229,63 @@ private void OnTimer(object? state)
                         counter.WritePayload((float)elapsed.TotalSeconds, _pollingIntervalInMilliseconds);
                     }
                     _timeStampSinceCollectionStarted = now;
-                }
-                else
-                {
-                    DisposeTimer();
+                    nextpollingTimeStamp = now + new TimeSpan(0, 0, 0, 0, _pollingIntervalInMilliseconds);
```

Comment:
This could drift over time if 'now' is measurably later than the desired wake up time. For example:
Timer enabled: 0.000
1st nextpollingTimeStamp: 1.000
Sleep returns: 1.001
OnTimer called: 1.003
now: 1.004
2nd nextPollingTimestamp: 2.004
…
3rd nextPollingTimestamp: 3.008
4th nextPollingTimestamp: 4.012

If a viewer is trying to match up the results we are reporting with some nominal time intervals the drift means that over larger periods of time there won't be enough results reported. Ie if we report a 1 sec interval result every 1.004 seconds then each hour there will 3585 datapoints and the viewer will need to do something weird if it wants to show 3600 datapoints.


How about:
```
do {
    nextpollingTimeStamp += _pollingIntervalInMilliseconds;
} while(nextpollingTimeStamp <= now)
```

Ideally that loop will only ever execute once, but if for whatever reason OnTimer() comes so slowly that we miss an entire polling interval then we'll skip ahead to the next one. It would still be drift in that case, but it is much less likely to occur and probably easier to deal with if occasionally one interval is surprisingly large rather than all intervals are a tiny bit large.